.github/workflows/test.yml

```diff
@@ -0,0 +1,37 @@
+name: Python Test Package
+
+on:
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [
+          # "3.6", Unsupported on ubuntu-latest (22.04)
+          # "3.7", Long series of scikit-learn failures
+          "3.8",
+          "3.9",
+          "3.10",
+          "3.11",
+          # "3.12", # Numba not available for Python 3.12: https://github.com/numba/numba/issues/9197
```

Comment:
Non-working version are commented with their reason for breaking

Reply:
It think it's fine to remove both 3.6 and 3.7 at this point though. Both have reached end of life officially.